proxy/proxy/httpserver.py

```diff
@@ -0,0 +1,171 @@
+                self.rfile.readline(65537)
+            )
+            logging.debug("New request: %s", self.raw_requestline)
+            if len(self.raw_requestline) > 65536:
+                self.requestline = ""  # pylint: disable=attribute-defined-outside-init
+                self.request_version = (  # pylint: disable=attribute-defined-outside-init
+                    ""
+                )
+                self.command = ""  # pylint: disable=attribute-defined-outside-init
+                self.send_error(HTTPStatus.REQUEST_URI_TOO_LONG)
+                return
+            if not self.raw_requestline:
+                self.close_connection = (  # pylint: disable=attribute-defined-outside-init
+                    True
+                )
+                return
+            if not self.parse_request():
+                # An error code has been sent, just exit
+                return
+            if self.command == "POST":
+                self.do_POST()
+            else:
+                url = f"{dst_protocol}://" + self.headers["Host"] + self.path
+                logging.debug("Passthrough none POST request: %s", url)
+                resp = requests.request(
+                    self.command, url, headers=self.headers, timeout=TIMEOUT
+                )
+                self.handle_response(resp)
+            self.wfile.flush()  # actually send the response if not already done.
+        except TimeoutError as e:
+            # a read or a write timed out.  Discard this connection
+            self.log_error("Request timed out: %r", e)
+            self.close_connection = (  # pylint: disable=attribute-defined-outside-init
+                True
+            )
+            return
+
+    def do_POST(self):  # pylint: disable=invalid-name
+        """do POST"""
+        logging.debug("POST")
+        url = f"{dst_protocol}://" + self.headers["Host"] + self.path
+        logging.debug("Passthrough POST request: %s", url)
+        content_length = int(self.headers["Content-Length"])
+        data = self.rfile.read(content_length)
+        job_request = self.path.find("/runtime/jobs") != -1
+        middleware_job_id = None
+        if job_request:
+            if "X-Qx-Client-Application" in self.headers:
+                logging.debug(
+                    "X-Qx-Client-Application: %s",
+                    self.headers["X-Qx-Client-Application"],
+                )
+                pos = self.headers["X-Qx-Client-Application"].find("middleware_job_id")
+                if pos != -1:
+                    middleware_job_id = self.headers["X-Qx-Client-Application"][
+                        pos
+                        + len("middleware_job_id/") : pos
+                        + len("middleware_job_id/")
+                        + 36
+                    ]
+                    logging.debug("Middleware Job ID: %s", middleware_job_id)
+
+        resp = None
+        while resp is None:
```

Comment:
Curious about this, why do we need to do a retry here? Did you find problems running the POST?

Reply:
It sometimes causes `ChunkedEncodingError`.  People seem retrying to workaround it.

Reply:
How do you see to add a limit of retries + a delay between retries? Thinking in:
- If the service had a problem not send requests indefinitely
- Letting some space between retries.